x/txfees/keeper/mempool-1559/code.go

```diff
@@ -10,51 +10,64 @@ import (
 	osmomath "github.com/osmosis-labs/osmosis/osmomath"
 )
 
-// Sections to this right now:
-// - Maintain something thats gets parsed from chain tx execution
-// update eipState according to that.
-// - Every time blockheight % 1000 = 0, reset eipState to default. (B/c this isn't committed on-chain, still gives us some consistency guarantees)
-// - Evaluate CheckTx/RecheckTx against this.
-//
-// 1000 blocks = almost 2 hours, maybe we need a smaller time for resets?
-// Lets say 500 blocks = 1 hour
-//
-// PROBLEMS: Currently, a node will throw out any tx that gets under its gas bound here.
-// :OOO We can just do this on checkTx not recheck
-//
-// Variables we can control for:
-// - fees paid per unit gas
-// - gas wanted per block (Ethereum)
-// - gas used and gas wanted difference
-// TODO: Change this percentage update time to be faster
-
-// TODO: Read this from config, can even make default 0, so this is only turned on by nodes who change it!
-// ALt: do that with an enable/disable flag. That seems likes a better idea
-
-var DefaultBaseFee = sdk.MustNewDecFromStr("0.0025")
-var MinBaseFee = sdk.MustNewDecFromStr("0.0025")
-var MaxBaseFee = sdk.MustNewDecFromStr("10")
-var TargetGas = int64(60_000_000)
-var MaxBlockChangeRate = sdk.NewDec(1).Quo(sdk.NewDec(16))
-var ResetInterval = int64(1000)
-var BackupFile = "eip1559state.json"
-var RecheckFeeConstant = int64(4)
+/*
+   This is the logic for the Osmosis implementation for EIP-1559 fee market,
+	 the goal of this code is to prevent spam by charging more for transactions when the network is busy.
+
+	 This logic does two things:
+   - Maintaining data parsed from chain transaction execution and updating eipState accordingly.
+   - Resetting eipState to default every ResetInterval (1000) block height intervals to maintain consistency.
+
+   Additionally:
+   - Periodically evaluating CheckTx and RecheckTx for compliance with these parameters.
+
+   Note: The reset interval is set to 1000 blocks, which is approximately 2 hours. Consider adjusting for a smaller time interval (e.g., 500 blocks = 1 hour) if necessary.
+
+   Challenges:
+   - Transactions falling under their gas bounds are currently discarded by nodes. This behavior can be modified for CheckTx, rather than RecheckTx.
+
+   Global variables stored in memory:
+   - DefaultBaseFee: Default base fee, initialized to 0.0025.
+   - MinBaseFee: Minimum base fee, initialized to 0.0025.
+   - MaxBaseFee: Maximum base fee, initialized to 10.
+   - TargetGas: Gas wanted per block, initialized to 60,000,000.
+   - MaxBlockChangeRate: The maximum block change rate, initialized to 1/16.
+   - ResetInterval: The interval at which eipState is reset, initialized to 1000 blocks.
+   - BackupFile: File for backup, set to "eip1559state.json".
+   - RecheckFeeConstant: A constant value for rechecking fees, initialized to 4.
```

Comment:
Should any of these be exposed via config?

Reply:
Not sure maybe @ValarDragon could input here?